File: task-1/src/main/java/ru/biluta/version5/Event.java
package ru.biluta.version5;

import lombok.Getter;

@Getter
public class Event implements Comparable<Event> {
    enum EventType {
        TASK_ARRIVAL,
        TASK_COMPLETION
    }

    private final EventType type;
    private final double time;
    private final Task task;
    private final Integer coreIndex;

    public Event(EventType type, double time, Task task, Integer coreIndex ) {
        this.type = type;
        this.time = time;
        this.task = task;
        this.coreIndex = coreIndex;
    }

    @Override
    public int compareTo(Event other) {
        return Double.compare(this.time, other.time);
    }
}

File: task2/src/main/kotlin/ru.biluta.task2/Event.kt
package main.kotlin.ru.biluta.task2

data class Event(

    val type: EventType,
    val time: Double,
    val task: Task,
    val coreIndex: Int?

): Comparable<Event> {

    override fun compareTo(other: Event): Int {
        return time.compareTo(other.time)
    }

}
